packages/material-ui/src/Slider/Slider.js

```diff
@@ -107,6 +107,9 @@ export const SliderRoot = experimentalStyled(
   },
   [`& .${sliderClasses.valueLabelLabel}`]: {
     color: theme.palette.primary.contrastText,
```

Comment:
This would work better, it allows to support any color, fixing a second bug. 

```suggestion
    color: theme.palette[styleProps.color].contrastText,
```

Reply:
Yes that's better. I quickly tried it against my test case and it looks good.